src/urllib3/http2.py

```diff
@@ -163,19 +235,60 @@ def getresponse(  # type: ignore[override]
             data=bytes(data),
         )
 
+    def request(  # type: ignore[override]
+        self,
+        method: str,
+        url: str,
+        body: _TYPE_BODY | None = None,
+        headers: typing.Mapping[str, str] | None = None,
+        *,
+        preload_content: bool = True,
+        decode_content: bool = True,
+        enforce_content_length: bool = True,
+        **kwargs: typing.Any,
+    ) -> None:
+        """Send an HTTP/2 request"""
+        if "chunked" in kwargs:
+            # TODO this is often present from upstream.
+            # raise NotImplementedError("`chunked` isn't supported with HTTP/2")
+            pass
+
+        if self.sock is not None:
+            self.sock.settimeout(self.timeout)
+
+        self.putrequest(method, url)
+
+        headers = headers or {}
+        for k, v in headers.items():
+            self.putheader(k, v)
+
+        if not self._has_header(b"user-agent"):
+            self.putheader(b"user-agent", _get_default_user_agent())
+
+        if body:
+            self.endheaders(message_body=body)
+            self.send(body)
+        else:
+            self.endheaders()
+
+    def _encode_request(self, request: str | bytes) -> bytes:
```

Comment:
Let's inline this to avoid more calls.

Reply:
Following CPython's lead here, again. Felt weird to leave a method behind that encoded ASCII.

It's only called by `putrequest`, so no benefit to breaking it out, either. Removed in my latest commit. 

https://github.com/python/cpython/blob/3.12/Lib/http/client.py#L1263